src/aws-api-mcp-server/awslabs/aws_api_mcp_server/scripts/download_latest_embeddings.py

```diff
@@ -0,0 +1,325 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/usr/bin/env python3
+
+import json
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tarfile
+import tempfile
+import zipfile
+from awscli.clidriver import __version__ as awscli_version
+from pathlib import Path
+from typing import Optional
+
+
+def run_command(command: list[str]) -> subprocess.CompletedProcess:
+    """Run a command and return the result, printing all output on error."""
+    print(f'Running: {" ".join(command)}')
+    try:
+        result = subprocess.run(command, capture_output=True, text=True, check=True)
+        if result.stderr:
+            print(f'STDERR: {result.stderr}')
+        return result
+    except subprocess.CalledProcessError as e:
+        print(f'Command failed with exit code {e.returncode}')
+        print(f'Command: {" ".join(e.cmd)}')
+        print(f'STDOUT: {e.stdout}')
+        print(f'STDERR: {e.stderr}')
+        raise
+
+
+def get_latest_artifact() -> Optional[dict]:
+    """Get the latest dist-aws-api-mcp-server artifact from GitHub."""
+    try:
+        repo = os.environ.get('GITHUB_REPOSITORY', 'awslabs/mcp')
+        owner = repo.split('/')[0]
+
+        cmd = [
+            'gh',
+            'api',
+            '--paginate',
+            '-H',
+            'Accept: application/vnd.github+json',
+            '-H',
+            'X-GitHub-Api-Version: 2022-11-28',
+            f'/repos/{owner}/mcp/actions/artifacts?name=dist-aws-api-mcp-server&per_page=100',
+        ]
+
+        result = run_command(cmd)
+
+        # Remove whitespace between objects, add commas, and wrap in []
+        json_objects = re.sub(r'}\s*{', '},{', result.stdout.strip())
+        json_array_str = f'[{json_objects}]'
+        try:
+            pages = json.loads(json_array_str)
+            all_artifacts = [artifact for page in pages for artifact in page.get('artifacts', [])]
+        except Exception as e:
+            print(f'Error parsing paginated JSON: {e}')
+            return None
+
+        # Sort by creation date and get the latest
+        if all_artifacts:
+            main_branch_artifacts = [
+                a for a in all_artifacts if a.get('workflow_run', {}).get('head_branch') == 'main'
```

Comment:
Is there a possibility the desired artifact is on a non-`main` branch? Pass in the ref as a parameter?

Reply:
The concern is artifacts must come from a `main` branch so updates must come from forks (won't be able to pull artifacts from alternative branches) - mark this as a future task.